### PR TITLE
Auth: Login and Register for APP and API.

### PR DESCRIPTION
**Pull request by samarv**
# COLABORATORS
- Jean S Ariza
- Samarvir Singh

# Description

Wire Frontend login and register components with the backend (API) Auth endpoints.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] APP were run in a local server. Interaction were done on the UI rendered in a browser.
- [x] DevTools were used to check HTTP calls and responses.
- [x] API were tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes THERE ARE NO TEST YET ON THIS BRANCH

### CODE REVIEW

File: backEnd/api/routes/Seekers.login.router.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const utils = require('../utils/authUtils');
+const Seeker = require('../models/Seeker/Seeker.model');
+// const Employers = require('../models/Employer/Employer.model');
```
Comment (d-vail) [inline backEnd/api/routes/Seekers.login.router.js]: This doesn't need to be here

File: backEnd/api/routes/Employer.Login.router.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+/**
+ * ROUTER HANDLERS: handle endpoints
+ */
+
+
+function loginEmployer(req, res, next) {
+  const { companyEmail, password } = req.body;
+  //console.log("email", companyEmail, "password", password)
```
Comment (d-vail) [inline backEnd/api/routes/Employer.Login.router.js]: This doesn't need to be here any more. We should keep console logs used for development out of the master branch

File: backEnd/api/routes/Employer.Login.router.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+              .status(422)
+              .json({ 'Bad Credentials': 'Please, check your credentials, there are some wrong data.' });
+
+          const jwt = utils.createToken({ _id, companyEmail });
+          res.status(200).json({ companyName, companyEmail, jwt });
+        })
+        .catch(e => {
+          console.log('error', e);
+          res.status(404).json({ e });
+        });
```
Comment (d-vail) [inline backEnd/api/routes/Employer.Login.router.js]: When I got an error from not having the secret key set, it didn't have any information in it. I just got back:

```
{
    "e": {}
}
```

**Comment by jesuarva** [comment]
# To pay attention to when reviewing this code
The employers model has a little changes it one of its properties. The field `companyName` had changed to `email`.
Do not run you MongoDB and try to test the Auth endpoints befor you have done the either one of the followings:
- Clean all your Employer data in the DB
- Create a new DB in your Mongo DB
 > You can change in `.env` cahnge MONGODB_URI='mongodb://localhost/developer-profiles'
 > to `MONGODB_URI='mongodb://localhost/developer-profiles2'`

Also add to `.env`:
```
 JWT_SECRET="your api secret goes here"
```

File: backEnd/api/models/Seeker/Seeker.model.js [hunk 4]
```diff
@@ -24,7 +24,8 @@ const seekerSchema = new Schema({
     required: [true, 'Password is required'],
     minlength: [8, 'Password must be at least 8 characters'],
     validate: {
-      validator: val => {
+      // prettier-ignore
+      validator: (val) => {
         return /^((?=\S*?[A-Z])(?=\S*?[a-z])(?=\S*?[0-9])(?=\S*?[!@#$%&]).{8,})\S$/.test(val);
       },
       message:
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply backEnd/api/models/Seeker/Seeker.model.js]: Got it, done.

File: backEnd/api/models/Seeker/Seeker.model.js [hunk 5]
```diff
@@ -36,7 +37,8 @@ const seekerSchema = new Schema({
     unique: true,
     required: [true, 'Email address is required'],
     validate: {
-      validator: val => {
+      // prettier-ignore
+      validator: (val) => {
         return /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/.test(val);
       },
       message: '{VALUE} is not a valid email address',
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply backEnd/api/models/Seeker/Seeker.model.js]: Done

File: backEnd/api/models/Seeker/Seeker.model.js [hunk 6]
```diff
@@ -97,15 +99,15 @@ const seekerSchema = new Schema({
...
     if (err) next(err);
     this.password = hash;
     next();
   });
 });
 
-seekerSchema.methods.isValidPassword = function(testPassword) {
+seekerSchema.methods.isValidPassword = function validatePassword(testPassword) {
   return bcrypt.compare(testPassword, this.password);
 };
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply backEnd/api/models/Seeker/Seeker.model.js]: Done

File: backEnd/api/utils/authUtils.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+
+const jwtSecret = process.env.JWT_SECRET;
+
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply backEnd/api/utils/authUtils.js]: Is already updated in the documentation

**Comment by jesuarva** [comment]
# BEFORE REVIEW
please read this previous comment: https://github.com/Lambda-School-Labs/CS10-Developer-Profiles/pull/31#issuecomment-421518889

**Comment by d-vail** [comment]
I am still seeing blank error messages from this https://github.com/Lambda-School-Labs/CS10-Developer-Profiles/pull/31#discussion_r217778727. I'm just going to make an issue for that so this can get merged to meet the milestone.

**Pull request merged**